### PR TITLE
fix(cubejs-client-core): Update types to work with TS v4.8 and above

### PR DESCRIPTION
**Pull request by sacummings91**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

[**Issue Reference this PR resolves**]

(https://github.com/cube-js/cube.js/issues/5446)

**Description of Changes Made (if issue reference is not provided)**

Small type changes to make tsc pass

### CODE REVIEW

**Comment by paveltiunov** [comment]
@sacummings91 Thanks for contributing it! cc @reify-thomas-smith

**Comment by paveltiunov** [comment]
@sacummings91 Could you please fix all failing tests?

File: packages/cubejs-client-core/index.d.ts [hunk 1]
```diff
@@ -816,7 +816,7 @@ declare module '@cubejs-client/core' {
     never;
 
   type QueryArrayRecordType<T extends DeeplyReadonly<Query[]>> =
-    T extends readonly [infer First, ...infer Rest]
+    T extends readonly [infer First extends DeeplyReadonly<Query>, ...infer Rest]
```
Comment (reify-thomas-smith) [inline packages/cubejs-client-core/index.d.ts:819]: Won't this break compatibility with older versions of TypeScript?

Comment (sacummings91) [reply packages/cubejs-client-core/index.d.ts:819]: I think you might be right about that, we can probably close this. I'm just going to set `"skipLibCheck: true"` in `tsconfig` instead as a work around

**Pull request closed without merging**